### PR TITLE
Op registration: math

### PR DESCRIPTION
**Pull request by asi1024**
Part of #6692.

### CODE REVIEW

File: chainerx_cc/chainerx/cuda/cuda_device/activation.cu [hunk 1]
```diff
@@ -32,23 +34,29 @@ struct IfLessElseASSAImpl {
...
-    CudaSetDeviceScope scope{index()};
-    VisitDtype(x_dtype, [&](auto x_pt) {
-        using In = typename decltype(x_pt)::type;
-        using CudaTypeIn = cuda_internal::DataType<In>;
-        VisitDtype(out.dtype(), [&](auto pt) {
-            using Out = typename decltype(pt)::type;
-            using CudaTypeOut = cuda_internal::DataType<Out>;
-            Elementwise<const In, const Out, Out>(
-                    IfLessElseASSAImpl<In, Out>{static_cast<CudaTypeIn>(x2), static_cast<CudaTypeOut>(pos)}, x1_cast, neg_cast, out);
+class CudaIfLessElseASSAOp : public IfLessElseASSAOp {
```
Comment (niboshi) [inline chainerx_cc/chainerx/cuda/cuda_device/activation.cu]: Enclose in `namespace {`.

Comment (asi1024) [reply chainerx_cc/chainerx/cuda/cuda_device/activation.cu]: Fixed.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by asi1024** [comment]
Resolved conflicts. Jenkins, test this please.

**Comment by niboshi** [comment]
Please resolve the conflict.

**Comment by niboshi** [comment]
@asi1024 Please merge instead of rebase when resolving conflict. Reviewing is much easier.
Also please refrain from triggering jenkins by yourself.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Hmm, please fix the conflict once again :(

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
LGTM

**Comment by hvy** [comment]
Removing `to-be-backported` as we decided not to backport non-critical PRs.

**Pull request merged**